packages/kbn-expandable-flyout/src/reducer.ts

```diff
@@ -130,7 +131,15 @@ export const reducer = createReducer(initialState, (builder) => {
     if (id in state.byId) {
       state.byId[id].right = right;
       state.byId[id].left = left;
-      state.byId[id].preview = preview ? [preview] : undefined;
+
+      // Only append preview when it does not match the last item in state.byId[id].preview
+      if (!deepEqual(preview, state.byId[id].preview?.at(-1))) {
+        if (state.byId[id].preview && preview) {
+          state.byId[id].preview?.push(preview);
+        } else {
+          state.byId[id].preview = preview ? [preview] : undefined;
+        }
+      }
```

Comment:
I think this would be slightly better:

```suggestion
      if (state.byId[id].preview && preview) {
        const previewIdenticalToLastOne = deepEqual(preview, state.byId[id].preview?.at(-1));
        // Only append preview when it does not match the last item in state.byId[id].preview
        if (!previewIdenticalToLastOne) {
          state.byId[id].preview?.push(preview);
        }
      } else {
        state.byId[id].preview = preview ? [preview] : undefined;
      }
```

We don't really need to check the deepEqual is `preview` is `null` or `undefined`, or also if `state.byId[id].preview` is null or undefined (which would mean we don't have any previews yet